src/types/duration.rs

```diff
@@ -12,15 +12,15 @@ pub struct DurationWrapper {
 impl DurationWrapper {
   #[napi]
   pub fn new(months: i32, days: i32, ns_bigint: BigInt) -> napi::Result<Self> {
-    let ns = ns_bigint.get_i64();
-    if !ns.1 {
+    let (ns_value,is_lossless) = ns_bigint.get_i64();
+    if !is_lossless {
       return Err(Error::new(
         Status::GenericFailure,
-        "Invalid use: Should not happen?",
+        "Nanoseconds cannot overflow i64",
       ));
     }
```

Comment:
Where does this guarantee come from? Is it some kind of runtime checks (where?) that prevents values out of `i64` range to be passed here?
In general, let's include such explanation upon each situation that we assume something and fail if it's not true.

Reply:
This is more of an expectation, than a guarantee. In the JS version, the nanoseconds are kept in i64, and if there is bigger value provided, it quietly changes the value. (not even in c style: by taking modulo 2^k). So by throwing this error we change a behavior in (improper use / undefined situation) of Duration type

Reply:
So maybe `Nanoseconds must not overflow i64` would convey the message better? As it's our expectation/limitation that we impose on the input and we forbid the user to pass something different.